modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/UpNextFragment.kt

```diff
@@ -517,6 +524,36 @@ class UpNextFragment : BaseFragment(), UpNextListener, UpNextTouchCallback.ItemT
         properties.putAll(props)
         analyticsTracker.track(event, properties)
     }
+
+    override fun onSortUpNext() {
+        val dialog = OptionsDialog()
+            .setTitle(getString(LR.string.sort_by))
+            .addTextOption(
+                titleId = LR.string.up_next_sort_added_to_up_next,
+                click = {
+                    viewLifecycleOwner.lifecycleScope.launch {
+                        playbackManager.upNextQueue.sortUpNextBy(UpNextSortType.ADDED_TO_UP_NEXT)
+                    }
```

Comment:
This is only a suggestion, I wonder if it would be better to push this into the view model.

```suggestion
                    upNextViewModel.sort(UpNextSortType.ADDED_TO_UP_NEXT)
```

and have the following in the view model

```kotlin
    fun sort(addedToUpNext: UpNextSortType) {
        viewModelScope.launch {
            playbackManager.upNextQueue.sortUpNextBy(addedToUpNext)
        }
    }
```

Reply:
Ummmmm phil in my opinion "add to upnext" should still be an option so you could hop between lists that's super powerful imo, 
I would probably just call it "custom default" or something like that
but yes the swipe or changing it should overwrite the existing 'custom default" & go to "custom default" option (add to up next would probably switch it too, which if you consider below I wouldn't be against but sort A-Z/Z-A should stick to those)

I would also love an option to sort into grouped feeds/subscribed so if you have multiple episodes from a feed it will group then together based off the existing feed order 🤔🤔 (as in the first episode the other episodes clings to that & then if you move then one of the episodes to the bottom it would move back to "custom default"